File: solutions/javascript/ReverseInteger/ReverseInteger.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    var max_int = 2147483647, min_int = -2147483648;
    var sign = x >= 0 ? true : false;
    x = parseFloat(Math.abs(x).toString().split('').reverse().join(''));
    x = sign ? x : -x;
    if (x > max_int || x < min_int) {
        return 0;
    }
    return x;
};

File: solutions/java/ReverseInteger/ReverseInteger.java
class Solution {
    public int reverse(int x) {
        boolean sign = x >= 0;
        String result = new StringBuffer(Math.abs((long)x) + "").reverse().toString();
        long l = Long.valueOf(result);
        l = sign ? l : -l;
        if (l > Integer.MAX_VALUE || l < Integer.MIN_VALUE) {
            return 0;
        }
        return (int)l;
    }
}